rest_api/sawtooth_rest_api/route_handlers.py

```diff
@@ -702,53 +709,39 @@ def _wrap_response(request, data=None, metadata=None, status=200):
 
     @classmethod
     def _wrap_paginated_response(
-        cls, request, response, controls, data, head=None
-    ):
+        cls, request, response, controls, data, head=None):
         """Builds the metadata for a pagingated response and wraps everying in
         a JSON encoded web.Response
         """
+        paging_response = response['paging']
         if head is None:
             head = response['head_id']
-        link = cls._build_url(request, head=head)
-
-        paging_response = response['paging']
-        total = paging_response['total_resources']
-        paging = {'total_count': total}
+        link =  cls._build_url(request,
+                               head=head,
+                               start=paging_response['start'],
+                               limit=paging_response['limit'])
 
+        paging = {}
         # If there are no resources, there should be nothing else in paging
-        if total == 0:
+        if paging_response.get("next") == "":
             return cls._wrap_response(
                 request,
                 data=data,
                 metadata={'head': head, 'link': link, 'paging': paging})
 
-        count = controls.get('count', len(data))
-        start = paging_response['start_index']
-        paging['start_index'] = start
+        limit = controls.get('limit')
+        start = controls.get("start")
+        paging["limit"] = limit
+        paging["start"] = start
```

Comment:
This need to go before the conditional on line 726, so that start and limit get returned on the last page as well.